packages/calypso-e2e/src/lib/pages/stats-page.ts

```diff
@@ -25,12 +71,143 @@ export class StatsPage {
 	 * @returns {Promise<void>} No return value.
 	 */
 	async clickTab( name: StatsTabs ): Promise< void > {
-		const dismissModalButton = this.page.getByRole( 'button', { name: 'Got it' } );
+		await this.dismissTooltip();
+		await clickNavTab( this.page, name );
 
-		if ( ( await dismissModalButton.count() ) > 0 ) {
-			await dismissModalButton.click();
-			await dismissModalButton.waitFor( { state: 'hidden' } );
+		// Wait for the expected URL scheme to load.
+		// Note, the Store tab is only available for Business and above plans.
+		if ( name === 'Traffic' ) {
+			await this.page.waitForURL( /stats\/day/ );
+		}
+		if ( name === 'Insights' ) {
+			await this.page.waitForURL( /stats\/insights/ );
+		}
+		if ( name === 'Subscribers' ) {
+			await this.page.waitForURL( /stats\/subscribers/ );
+		}
+		if ( name === 'Store' ) {
+			await this.page.waitForURL( /store\/stats\/orders/ );
```

Comment:
Despite the fact that the endpoint schema doesn't line up exactly with the tab name, there's probably a one line way to do this.

That said, this approach is probably more maintainable and legible than a fancy one-liner.